src/ctl.c

```diff
@@ -3083,3 +3085,44 @@ experimental_hooks_remove_ctl(tsd_t *tsd, const size_t *mib, size_t miblen,
 label_return:
 	return ret;
 }
+
+static int
+experimental_frag_ctl(tsd_t *tsd, const size_t *mib, size_t miblen,
+    void *oldp, size_t *oldlenp, void *newp, size_t newlen) {
+	assert(oldp != NULL && newp != NULL);
+
+	/* trim *oldlenp and newlen so that:
+	 * (a) the larger is reduced to fit the smaller
+	 * (b) both are trimmed to a multiple of the unit size (if not yet)
+	 * rely on compiler to upgrade multiplication / division to shifts
+	 */
+	if (*oldlenp / (2 * sizeof(uint16_t)) < newlen / sizeof(void *)) {
+		newlen = *oldlenp / (2 * sizeof(uint16_t)) * sizeof(void *);
+		*oldlenp -= *oldlenp % (2 * sizeof(uint16_t));
+	} else {
+		*oldlenp = newlen / sizeof(void *) * (2 * sizeof(uint16_t));
+		newlen -= newlen % sizeof(void *);
+	}
+
+	/* each pointer will output two short entries for the extent it
+	 * resides in, where the first one is the number of free regions and
+	 * the second one is the total number of regions
+	 */
+	void * const newend = newp + newlen;
```

Comment:
It looks this is causing issues in Windows CI: https://ci.appveyor.com/project/jasone/jemalloc/builds/23165922/job/q69f6p9pi9ijulq1